doc/developers/auth/jwt.rst

```diff
@@ -6,41 +6,78 @@ JWT (`JSON Web Token <https://tools.ietf.org/html/rfc7519>`_) bearer authenticat
+Setting up a token issuer with manual key refresh
+-------------------------------------------------
 
 Before adding public token signing keys to a running CCF network, the IdP has to be stored as token issuer with a ``set_jwt_issuer`` proposal:
 
 .. code-block:: bash
 
     $ cat issuer.json
     {
-      "issuer": "https://login.microsoftonline.com/common/v2.0"
+      "issuer": "my-issuer",
+      "auto_refresh": false
     }
     $ python -m ccf.proposal_generator set_jwt_issuer issuer.json
 
+The ``issuer`` field is an arbitrary identifier and can be used during token validation to differentiate between multiple issuers.
+
 Note that ``issuer.json`` has some additional optional fields for more advanced scenarios.
 See :ref:`developers/auth/jwt:Advanced issuer configuration` for details.
 
 After this proposal is accepted, signing keys for an issuer can be updated with a ``set_jwt_public_signing_keys`` proposal:
 
 .. code-block:: bash
 
-    $ curl "https://login.microjsononline.com/common/discovery/v2.0/keys" -o jwks.json
-    $ ISSUER="https://login.microsoftonline.com/common/v2.0"
+    $ ISSUER="my-issuer"
     $ python -m ccf.proposal_generator set_jwt_public_signing_keys $ISSUER jwks.json
 
+``jwks.json`` contains the signing keys as JWKS (`JSON Web Key Set <https://tools.ietf.org/html/rfc7517>`_) document. 
+
+Setting up a token issuer with automatic key refresh
+----------------------------------------------------
+
+Most IdPs follow the OpenID Connect Discovery specification and publish their public token signing keys at a well-known location.
+In such cases, token issuers in CCF can be configured to automatically refresh their keys.
+
+The following extra conditions must be true compared to setting up an issuer with manual key refresh:
+
+- The ``issuer`` must be an OpenID Connect issuer URL, for example ``https://login.microsoftonline.com/common/v2.0``. During auto-refresh, the keys are fetched from that URL by appending ``/.well-known/openid-configuration``.
+- A CA certificate for the issuer URL must be stored so that the TLS connection to the IdP can be validated during key refresh.
+
+The CA certificate is stored with a ``set_ca_cert`` proposal:
+
+.. code-block:: bash
+
+    $ python -m ccf.proposal_generator set_ca_cert jwt_ms cacert.pem
+
+Now the issuer can be created with auto-refresh enabled:
+
+.. code-block:: bash
+
+    $ cat issuer.json
+    {
+      "issuer": "https://login.microsoftonline.com/common/v2.0",
+      "auto_refresh": true,
+      "ca_cert_name": "jwt_ms"
+    }
+    $ python -m ccf.proposal_generator set_jwt_issuer issuer.json
+
+.. note::
+
+    The key refresh interval is set via the ``--jwt-key-refresh-interval-s <seconds>`` CLI option for ``cchost``, where the default is 30 min.
+
+.. warning::
+
+    Currently, no initial automatic key refresh takes place when adding a new token issuer.
```

Comment:
That's quite a strange limitation, something we are planning to address?

Reply:
Definitely, it's just a bit more tricky since a one-off task has to be scheduled. Should I mention that this is a temporary limitation?

Reply:
Extended the warning.